Support for recorded and paused time

FJBDev opened the pull request:
Changes per code review suggestions in previous [PR](https://github.com/wolfgang-ch/mytourbook/pull/222)

wolfgang-ch commented:
@FJBDev Let me know when I should do a review.

FJBDev commented:
> @FJBDev Let me know when I should do a review.

You can do a review :-)

wolfgang-ch commented:
The code formatting do not join lines, so there can be "line breaks".

In the beginning the join line feature was tuned on but there was an issue (which I can't remember) that I turned it off. Maybe in later Eclipse versions this can be setup more granularly as I also hate to join the lines manually.

This is not an issue to not merge it into `main` branch

![mt-join-lines](https://user-images.githubusercontent.com/1283445/93012354-14289d80-f5a0-11ea-939c-e6df217a017c.png)

![mt-never-join-lines](https://user-images.githubusercontent.com/1283445/93012358-17bc2480-f5a0-11ea-926c-48836f756fc4.png)

wolfgang-ch commented on bundles/net.tourbook/src/net/tourbook/data/TourData.java:
```diff
@@ -1685,8 +1668,23 @@
+   @Transient
+   public long[]           pausedTime_Start;
 
-// SET_FORMATTING_ON
```
Code formatting should be turned on after all fields are defined, otherwise the following code is not formatted

FJBDev replied:
Eclipse removed it without asking me. I just put it back.

wolfgang-ch replied:
When refactoring code then Eclipse can remove comment lines which are above the refactored code, this is my experience.

wolfgang-ch commented:
Since a while I'm using the state contant name also for the state key, this is easier to use by just copy/paste it :-)

![mt-pref-values](https://user-images.githubusercontent.com/1283445/93012593-fbb98280-f5a1-11ea-86b3-8e15f56b6e18.png)

wolfgang-ch commented:
When using `UI.EMPTY_STRING` then it is better to use it from `net.tourbook.common.UI` because when using other `net.tourbook.common.UI` resources then the fully qualified name for `UI` will be used

![mt-ui-common](https://user-images.githubusercontent.com/1283445/93012727-da0ccb00-f5a2-11ea-9e99-a4830ba0c555.png)

wolfgang-ch commented:
`tourRecordingTime` is also renamed in the table `TourCompared`, I think this was not intended as the database update method do not reflect this renaming. I've not tested it but I think, MT will fail when it is accessing this table.


![mt-tour-compared-2](https://user-images.githubusercontent.com/1283445/93012812-6fa85a80-f5a3-11ea-811b-0498149b581b.png)

![mt-tour-compared](https://user-images.githubusercontent.com/1283445/93012808-6b7c3d00-f5a3-11ea-9e65-6bd1170b2680.png)

wolfgang-ch reviewed (commented):
After 2.5h of code review, I need to rest my brain as I'm getting tired. This is the reason why I can't work 100% after my bicycle accident. I think what makes me tired are the many "context" switches during the review as there are so many different files to look at.

Here you can see my current review location, I started from top to bottom and collapse all which is reviewed

![mt-review](https://user-images.githubusercontent.com/1283445/93012924-64a1fa00-f5a4-11ea-8e8c-942142674719.png)

wolfgang-ch commented on bundles/net.tourbook.device.garmin/src/net/tourbook/device/garmin/GarminSAXHandler.java:
```diff
@@ -111,69 +109,71 @@
+   private final List<Long>        _pausedTime_Start  = new ArrayList<>();
+   private List<Long>              _pausedTime_End    = new ArrayList<>();
+   private int                     _currentPauseIndex = -1;
+   private boolean                 _isPreviousLapAPause;
```
`_isPreviousLapAPause` Typo?

FJBDev replied:
No that looks good to me. This variable is used to indicate if a previously found lap was a pause.

wolfgang-ch replied:
After reading you comment, this field name is OK it just looked strange when I first saw it.

wolfgang-ch commented:
> `tourRecordingTime` is also renamed in the table `TourCompared`, I think this was not intended as the database update method do not reflect this renaming. I've not tested it but I think, MT will fail when it is accessing this table.
> 
> ![mt-tour-compared-2](https://user-images.githubusercontent.com/1283445/93012812-6fa85a80-f5a3-11ea-811b-0498149b581b.png)


This is the corresponding error when expanding a year in the ref tour view

![mt-sql-exception](https://user-images.githubusercontent.com/1283445/93086369-541e7c00-f697-11ea-911a-a4fff291f605.png)

wolfgang-ch reviewed (commented):
The code review is now done with 1 major issue, I'll do a functional review when the code is OK

FJBDev commented:
> 
> 
> The code review is now done with 1 major issue, I'll do a functional review when the code is OK

OK, i will fix the major issue and the other code suggestions quickly.

FJBDev commented:
> 
> 
> Since a while I'm using the state contant name also for the state key, this is easier to use by just copy/paste it :-)
> 
> ![mt-pref-values](https://user-images.githubusercontent.com/1283445/93012593-fbb98280-f5a1-11ea-86b3-8e15f56b6e18.png)

Fixed

FJBDev commented:
> 
> 
> The code formatting do not join lines, so there can be "line breaks".
> 
> In the beginning the join line feature was tuned on but there was an issue (which I can't remember) that I turned it off. Maybe in later Eclipse versions this can be setup more granularly as I also hate to join the lines manually.
> 
> This is not an issue to not merge it into `main` branch
> 
> ![mt-join-lines](https://user-images.githubusercontent.com/1283445/93012354-14289d80-f5a0-11ea-939c-e6df217a017c.png)
> 
> ![mt-never-join-lines](https://user-images.githubusercontent.com/1283445/93012358-17bc2480-f5a0-11ea-926c-48836f756fc4.png)

Fixed

FJBDev commented:
> 
> 
> When using `UI.EMPTY_STRING` then it is better to use it from `net.tourbook.common.UI` because when using other `net.tourbook.common.UI` resources then the fully qualified name for `UI` will be used
> 
> ![mt-ui-common](https://user-images.githubusercontent.com/1283445/93012727-da0ccb00-f5a2-11ea-9e99-a4830ba0c555.png)

Fixed

FJBDev commented:
> 
> 
> > `tourRecordingTime` is also renamed in the table `TourCompared`, I think this was not intended as the database update method do not reflect this renaming. I've not tested it but I think, MT will fail when it is accessing this table.
> > ![mt-tour-compared-2](https://user-images.githubusercontent.com/1283445/93012812-6fa85a80-f5a3-11ea-811b-0498149b581b.png)
> 
> This is the corresponding error when expanding a year in the ref tour view
> 
> ![mt-sql-exception](https://user-images.githubusercontent.com/1283445/93086369-541e7c00-f697-11ea-911a-a4fff291f605.png)

I have 

> 
> 
> The code review is now done with 1 major issue, I'll do a functional review when the code is OK

Ok, I have a fixed this 1 major issue [here](https://github.com/wolfgang-ch/mytourbook/pull/227).

FJBDev commented:
> 
> 
> `tourRecordingTime` is also renamed in the table `TourCompared`, I think this was not intended as the database update method do not reflect this renaming. I've not tested it but I think, MT will fail when it is accessing this table.
> 
> ![mt-tour-compared-2](https://user-images.githubusercontent.com/1283445/93012812-6fa85a80-f5a3-11ea-811b-0498149b581b.png)
> 
> ![mt-tour-compared](https://user-images.githubusercontent.com/1283445/93012808-6b7c3d00-f5a3-11ea-9e65-6bd1170b2680.png)

I have renamed the field in the database. I had forgotten to do so.

FJBDev commented:
> 
> 
> After 2.5h of code review, I need to rest my brain as I'm getting tired. This is the reason why I can't work 100% after my bicycle accident. I think what makes me tired are the many "context" switches during the review as there are so many different files to look at.
> 
> Here you can see my current review location, I started from top to bottom and collapse all which is reviewed
> 
> ![mt-review](https://user-images.githubusercontent.com/1283445/93012924-64a1fa00-f5a4-11ea-8e8c-942142674719.png)

Indeed, lots of changes and work for this feature. I started several weeks ago now :-)
